modules/document_repository/ajax/documentEditUpload.php

```diff
@@ -52,34 +44,39 @@
     || $userSingleton->hasPermission('document_repository_delete')
 ) {
     if ($action == 'upload') {
-        $puser      = $_POST['user'];
-        $category   = $_POST['category'];
+        $category   = $_POST['category']; // required
         $site       = $_POST['site']       !== '' ? $_POST['site'] : null;
         $instrument = $_POST['instrument'] !== '' ? $_POST['instrument'] : null;
         $pscid      = $_POST['pscid']      !== '' ? $_POST['pscid'] : null;
         $visit      = $_POST['visit']      !== '' ? $_POST['visit'] : null;
         $comments   = $_POST['comments']   !== '' ? $_POST['comments'] : null;
         $version    = $_POST['version']    !== '' ? $_POST['version'] : null;
 
-        $fileSize = $_FILES["file"]["size"];
-        $fileName = $_FILES["file"]["name"];
-        $fileType = end((explode(".", $fileName)));
-
-        // __DIR__ is the document_repository ajax directory
-        // when this script is executing. Go up a level to the
-        // document_repository module directory, and use a
-        // user_uploads directory as a base for user uploads
-        $base_path = __DIR__ . "/../user_uploads/";
-        $fileBase  = $puser . "/" . $fileName;
-
-        if (!file_exists($base_path . $puser)) {
-            mkdir($base_path . $puser, 0777);
+        $fileSize = $_FILES['file']['size'];
+        $fileName = $_FILES['file']['name'];
+        $fileType = pathinfo($_FILES['file']['name'], PATHINFO_EXTENSION);
+
+        $uploadPath = "$base/modules/document_repository/user_uploads/$name/";
+        $fullPath   = $uploadPath . $fileName;
+
+        if (!is_writable($uploadPath)) {
+            if (file_exists($uploadPath)) {
+                http_response_code(403);
+                error_log("Could not write to $uploadPath. Check permissions");
+                exit;
+            }
+            error_log(
+                'Creating document repository upload folder for ' .
+                'user ' . $name
+            );
```

Comment:
from my understanding, is_writable returns true if the filename exists AND is writable. in the happy path case that !file_exists, $uploadPath can still be not writable. wouldn't this cause an error?

Reply:
No I think the logic is good:
1. Check if file `is_writable`
2. If the file is NOT writable, then either: the file does not exist OR we have insufficient permissions
3. Check if the file exists
4. If the file does exist, but is not writable, give a forbidden error
5. If the file does not exist, create one.

Reply:
but what if the file does not exist AND we have insufficient permissions? wouldn't that case also enter the !is_writable($uploadPath) if statement?

Reply:
>if the file does not exist AND we have insufficient permissions
`is_writable` checks one file/folder to see whether the current user (in this case, `www-data`/`apache`) has write permissions for that file. 
If the file doesn't exist it doesn't make sense to speak of what options are set on it.

If you're asking whether that user has permissions to write to `user_uploads/` that file is owned by the web user [by default](https://github.com/aces/Loris/blob/030d1c08989e7b94232d6996253bd8ffabb87199/tools/install.sh#L158).

This doesn't intersect in any way with the permissions associated with loris front-end users; it checks only whether the Linux system user can do file operations.

Reply:
i see, this makes sense